File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        zi, i = 0, 0

        while zi <= i and i < len(nums):
            if nums[zi] != 0:
                while zi < len(nums) and nums[zi] != 0:
                    zi += 1
                    i = zi + 1
            if i < len(nums) and nums[i] != 0:
                nums[zi] = nums[i]
                nums[i] = 0
                zi += 1            
            i += 1
        

File: 0283-move-zeroes/0283-move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {

    let i = 0;
    let j = 1;

    while (i < j && j < nums.length) {
        if (nums[i] === 0 && nums[j] !== 0) {
            const temp = nums[i];
            nums[i] = nums[j];
            nums[j] = temp;
        }
        if (nums[i] !== 0) {
            i++;
            j = i + 1;
        }
        if (nums[j] === 0) j++;
    }

    
};